Bumping versions of transitive dependencies leads to segfault in `pytest`

To replicate, run:

```sh
uv lock --upgrade
```

Then:

```sh
pytest
```

On my machine I get a nasty error which looks like the below. It looks like it might be caused by the `pytest-qt` package.

```
tests/gui/hardware_set/test_hardware_sets_view.py::test_on_device_open_end
[gw4] [ 96%] PASSED tests/gui/hardware_set/test_hardware_sets_view.py::test_on_device_open_end
tests/gui/hardware_set/test_hardware_sets_view.py::test_on_device_closed Fatal Python error: Segmentation fault

Thread 0x00007f96d3dff6c0 [python3] (most recent call first):
  File "/home/alex/code/FROG/.venv/lib/python3.14/site-packages/execnet/gateway_base.py", line 534 in read
  File "/home/alex/code/FROG/.venv/lib/python3.14/site-packages/execnet/gateway_base.py", line 567 in from_io
  File "/home/alex/code/FROG/.venv/lib/python3.14/site-packages/execnet/gateway_base.py", line 1160 in _thread_receiver
  File "/home/alex/code/FROG/.venv/lib/python3.14/site-packages/execnet/gateway_base.py", line 341 in run
  File "/home/alex/code/FROG/.venv/lib/python3.14/site-packages/execnet/gateway_base.py", line 411 in _perform_spawn

Current thread 0x00007f96d56d0bc0 [python3] (most recent call first):
  File "/home/alex/code/FROG/src/frog/gui/hardware_set/hardware_sets_view.py", line 167 in __init__
  File "/home/alex/code/FROG/tests/gui/hardware_set/test_hardware_sets_view.py", line 42 in hw_control
  File "/usr/lib/python3.14/unittest/mock.py", line 1432 in patched
  File "/home/alex/code/FROG/.venv/lib/python3.14/site-packages/_pytest/fixtures.py", line 930 in call_fixture_func
  File "/home/alex/code/FROG/.venv/lib/python3.14/site-packages/_pytest/fixtures.py", line 1196 in pytest_fixture_setup
  File "/home/alex/code/FROG/.venv/lib/python3.14/site-packages/pluggy/_callers.py", line 121 in _multicall
  File "/home/alex/code/FROG/.venv/lib/python3.14/site-packages/pluggy/_manager.py", line 120 in _hookexec
  File "/home/alex/code/FROG/.venv/lib/python3.14/site-packages/pluggy/_hooks.py", line 512 in __call__
  File "/home/alex/code/FROG/.venv/lib/python3.14/site-packages/_pytest/fixtures.py", line 1128 in execute
  File "/home/alex/code/FROG/.venv/lib/python3.14/site-packages/_pytest/fixtures.py", line 640 in _get_active_fixturedef
  File "/home/alex/code/FROG/.venv/lib/python3.14/site-packages/_pytest/fixtures.py", line 549 in getfixturevalue
  File "/home/alex/code/FROG/.venv/lib/python3.14/site-packages/_pytest/fixtures.py", line 720 in _fillfixtures
  File "/home/alex/code/FROG/.venv/lib/python3.14/site-packages/_pytest/python.py", line 1674 in setup
  File "/home/alex/code/FROG/.venv/lib/python3.14/site-packages/_pytest/runner.py", line 514 in setup
  File "/home/alex/code/FROG/.venv/lib/python3.14/site-packages/_pytest/runner.py", line 164 in pytest_runtest_setup
  File "/home/alex/code/FROG/.venv/lib/python3.14/site-packages/pluggy/_callers.py", line 121 in _multicall
  File "/home/alex/code/FROG/.venv/lib/python3.14/site-packages/pluggy/_manager.py", line 120 in _hookexec
  File "/home/alex/code/FROG/.venv/lib/python3.14/site-packages/pluggy/_hooks.py", line 512 in __call__
  File "/home/alex/code/FROG/.venv/lib/python3.14/site-packages/_pytest/runner.py", line 246 in <lambda>
  File "/home/alex/code/FROG/.venv/lib/python3.14/site-packages/_pytest/runner.py", line 344 in from_call
  File "/home/alex/code/FROG/.venv/lib/python3.14/site-packages/_pytest/runner.py", line 245 in call_and_report
  File "/home/alex/code/FROG/.venv/lib/python3.14/site-packages/_pytest/runner.py", line 130 in runtestprotocol
  File "/home/alex/code/FROG/.venv/lib/python3.14/site-packages/_pytest/runner.py", line 117 in pytest_runtest_protocol
  File "/home/alex/code/FROG/.venv/lib/python3.14/site-packages/pluggy/_callers.py", line 121 in _multicall
  File "/home/alex/code/FROG/.venv/lib/python3.14/site-packages/pluggy/_manager.py", line 120 in _hookexec
  File "/home/alex/code/FROG/.venv/lib/python3.14/site-packages/pluggy/_hooks.py", line 512 in __call__
  File "/home/alex/code/FROG/.venv/lib/python3.14/site-packages/xdist/remote.py", line 227 in run_one_test
  File "/home/alex/code/FROG/.venv/lib/python3.14/site-packages/xdist/remote.py", line 206 in pytest_runtestloop
  File "/home/alex/code/FROG/.venv/lib/python3.14/site-packages/pluggy/_callers.py", line 121 in _multicall
  File "/home/alex/code/FROG/.venv/lib/python3.14/site-packages/pluggy/_manager.py", line 120 in _hookexec
  File "/home/alex/code/FROG/.venv/lib/python3.14/site-packages/pluggy/_hooks.py", line 512 in __call__
  File "/home/alex/code/FROG/.venv/lib/python3.14/site-packages/_pytest/main.py", line 343 in _main
  File "/home/alex/code/FROG/.venv/lib/python3.14/site-packages/_pytest/main.py", line 289 in wrap_session
  File "/home/alex/code/FROG/.venv/lib/python3.14/site-packages/_pytest/main.py", line 336 in pytest_cmdline_main
  File "/home/alex/code/FROG/.venv/lib/python3.14/site-packages/pluggy/_callers.py", line 121 in _multicall
  File "/home/alex/code/FROG/.venv/lib/python3.14/site-packages/pluggy/_manager.py", line 120 in _hookexec
  File "/home/alex/code/FROG/.venv/lib/python3.14/site-packages/pluggy/_hooks.py", line 512 in __call__
  File "/home/alex/code/FROG/.venv/lib/python3.14/site-packages/xdist/remote.py", line 427 in <module>
  File "/home/alex/code/FROG/.venv/lib/python3.14/site-packages/execnet/gateway_base.py", line 1291 in executetask
  File "/home/alex/code/FROG/.venv/lib/python3.14/site-packages/execnet/gateway_base.py", line 341 in run
  File "/home/alex/code/FROG/.venv/lib/python3.14/site-packages/execnet/gateway_base.py", line 411 in _perform_spawn
  File "/home/alex/code/FROG/.venv/lib/python3.14/site-packages/execnet/gateway_base.py", line 389 in integrate_as_primary_thread
  File "/home/alex/code/FROG/.venv/lib/python3.14/site-packages/execnet/gateway_base.py", line 1273 in serve
  File "/home/alex/code/FROG/.venv/lib/python3.14/site-packages/execnet/gateway_base.py", line 1806 in serve
  File "<string>", line 8 in <module>
  File "<string>", line 1 in <module>

Current thread's C stack trace (most recent call first):
  Binary file "/usr/lib/libpython3.14.so.1.0", at _Py_DumpStack+0x4d [0x7f96d51409bb]
  Binary file "/usr/lib/libpython3.14.so.1.0", at +0x145321 [0x7f96d5145321]
  Binary file "/usr/lib/libc.so.6", at +0x3e4d0 [0x7f96d4e4d4d0]
  Binary file "/home/alex/code/FROG/.venv/lib/python3.14/site-packages/PySide6/libpyside6.abi3.so.6.10", at +0x2517a [0x7f96d1ab917a]
  Binary file "/home/alex/code/FROG/.venv/lib/python3.14/site-packages/PySide6/libpyside6.abi3.so.6.10", at +0x25710 [0x7f96d1ab9710]
  Binary file "/home/alex/code/FROG/.venv/lib/python3.14/site-packages/PySide6/libpyside6.abi3.so.6.10", at _ZN6PySide22qobjectConnectCallbackEP7QObjectPKcP7_objectN2Qt14ConnectionTypeE+0xc7 [0x7f96d1aa9cf8]
  Binary file "/home/alex/code/FROG/.venv/lib/python3.14/site-packages/PySide6/libpyside6.abi3.so.6.10", at +0x20c09 [0x7f96d1ab4c09]
  Binary file "/usr/lib/libpython3.14.so.1.0", at +0x1b8b85 [0x7f96d51b8b85]
  Binary file "/usr/lib/libpython3.14.so.1.0", at PyObject_Vectorcall+0x5d [0x7f96d517c72d]
  Binary file "/usr/lib/libpython3.14.so.1.0", at _PyEval_EvalFrameDefault+0x1373 [0x7f96d51917c3]
  Binary file "/usr/lib/libpython3.14.so.1.0", at +0x1c6e3b [0x7f96d51c6e3b]
  Binary file "/usr/lib/libpython3.14.so.1.0", at _PyObject_MakeTpCall+0x48b [0x7f96d517869b]
  Binary file "/usr/lib/libpython3.14.so.1.0", at _PyEval_EvalFrameDefault+0xa40a [0x7f96d519a85a]
  Binary file "/usr/lib/libpython3.14.so.1.0", at +0x1be86c [0x7f96d51be86c]
  Binary file "/usr/lib/libpython3.14.so.1.0", at +0x274d31 [0x7f96d5274d31]
  Binary file "/usr/lib/libpython3.14.so.1.0", at +0x2d0b21 [0x7f96d52d0b21]
  Binary file "/usr/lib/libpython3.14.so.1.0", at _PyObject_MakeTpCall+0x46c [0x7f96d517867c]
  Binary file "/usr/lib/libpython3.14.so.1.0", at _PyEval_EvalFrameDefault+0x6a6d [0x7f96d5196ebd]
  Binary file "/usr/lib/libpython3.14.so.1.0", at +0x1be86c [0x7f96d51be86c]
  Binary file "/usr/lib/libpython3.14.so.1.0", at +0x274d31 [0x7f96d5274d31]
  Binary file "/usr/lib/libpython3.14.so.1.0", at +0x2d0b21 [0x7f96d52d0b21]
  Binary file "/usr/lib/libpython3.14.so.1.0", at +0x1d1210 [0x7f96d51d1210]
  Binary file "/usr/lib/libpython3.14.so.1.0", at _PyEval_EvalFrameDefault+0x4b44 [0x7f96d5194f94]
  Binary file "/usr/lib/libpython3.14.so.1.0", at +0x1be86c [0x7f96d51be86c]
  Binary file "/usr/lib/libpython3.14.so.1.0", at +0x274d31 [0x7f96d5274d31]
  Binary file "/usr/lib/libpython3.14.so.1.0", at +0x2d0b21 [0x7f96d52d0b21]
  Binary file "/usr/lib/libpython3.14.so.1.0", at _PyObject_MakeTpCall+0x46c [0x7f96d517867c]
  Binary file "/usr/lib/libpython3.14.so.1.0", at _PyEval_EvalFrameDefault+0x6a6d [0x7f96d5196ebd]
  Binary file "/usr/lib/libpython3.14.so.1.0", at +0x202f96 [0x7f96d5202f96]
  Binary file "/usr/lib/libpython3.14.so.1.0", at _PyEval_EvalFrameDefault+0x4b44 [0x7f96d5194f94]
  Binary file "/usr/lib/libpython3.14.so.1.0", at +0x1be86c [0x7f96d51be86c]
  Binary file "/usr/lib/libpython3.14.so.1.0", at +0x274d31 [0x7f96d5274d31]
  <truncated rest of calls>

Extension modules: shiboken6.Shiboken, PySide6.QtCore, PySide6.QtGui, PySide6.QtWidgets, PySide6.QtTest, yaml._yaml, numpy._core._multiarray_umath, numpy.linalg._umath_linalg, PySide6.QtNetwork, PySide6.QtPrintSupport, PySide6.QtWebChannel, PySide6.QtWebEngineCore, PySide6.QtWebEngineWidgets, PIL._imaging, kiwisolver._cext (total: 15)

[gw1] node down: Not properly terminated
[gw1] [ 96%] FAILED tests/gui/hardware_set/test_hardware_sets_view.py::test_init[None]
```

--- a/pyproject.toml
+++ b/pyproject.toml
@@ -12,9 +12,7 @@ authors = [
 ]
 requires-python = ">=3.14"
 dependencies = [
-    # PySide6 v6.10 not working with pytest-qt:
-    #   https://github.com/ImperialCollegeLondon/FROG/issues/909
-    "PySide6<6.10.0,>=6.9.0",
+    "pyside6<7.0.0,>=6.11.1",
     "matplotlib>=3.11.0,<4.0.0",
     "platformdirs>=4.10.0,<5.0.0",
     "pypubsub<5.0.0,>=4.0.3",

--- a/src/frog/gui/hardware_set/device_view.py
+++ b/src/frog/gui/hardware_set/device_view.py
@@ -267,7 +267,7 @@ class DeviceTypeControl(QGroupBox):
 
         # For some reason we also have to hide the widget else it appears over the
         # others
-        layout.takeAt(1).widget().hide()
+        layout.takeAt(1).widget().hide()  # type: ignore[union-attr]
 
         # Add the widget for the newly selected parameter if needed
         widget = self.current_device_type_widget

--- a/src/frog/gui/stepper_motor_view.py
+++ b/src/frog/gui/stepper_motor_view.py
@@ -108,10 +108,10 @@ class StepperMotorControl(DevicePanel):
 
             # Also check the corresponding preset button
             preset_upper = preset.upper()
-            btn = next(
+            preset_btn = next(
                 btn for btn in self.button_group.buttons() if btn.text() == preset_upper
             )
-            btn.setChecked(True)
+            preset_btn.setChecked(True)
         elif btn := self.button_group.checkedButton():
             # This angle isn't a preset. If there is a button already checked, uncheck
             # it.

--- a/tests/gui/test_sensors_panel.py
+++ b/tests/gui/test_sensors_panel.py
@@ -53,7 +53,7 @@ def test_get_reading_lineedit() -> None:
     )
     assert isinstance(reading, QLineEdit)
     assert panel._reading_layout.rowCount() == 1
-    reading_name_label = panel._reading_layout.itemAt(0).widget()
+    reading_name_label = panel._reading_layout.itemAt(0).widget()  # type: ignore[union-attr]
     assert isinstance(reading_name_label, QLabel)
     assert reading_name_label.text() == "Quantity1"
 

